packages/app/src/lib/workers/backend/worker.ts

```diff
@@ -0,0 +1,589 @@
+        ] of this.sqlite.liveQueries.entries()) {
+          const channel = this.sqlite.createLiveQueryChannel(port);
+          this.sqlite.createLiveQuery(id, channel.port2, statement);
+        }
+      },
+      async ping() {
+        console.log("Backend: Ping received");
+        return {
+          timestamp: Date.now(),
+        };
+      },
+      enableLogForwarding: () => this.enableLogForwarding(),
+      disableLogForwarding: () => this.disableLogForwarding(),
+      createStream: async (ulid, moduleId, moduleUrl, params) =>
+        this.client.createStream(ulid, moduleId, moduleUrl, params),
+      sendEvent: async (streamId: string, event: EventType) => {
+        await this.client.sendEvent(streamId, event);
+      },
+      sendEventBatch: async (streamId, payloads) => {
+        await this.client.sendEventBatch(streamId, payloads);
+      },
+      fetchEvents: async (streamId, offset, limit) =>
+        this.client.fetchEvents(streamId, offset, limit),
+      previewSpace: async (_streamId) => {
+        await this.sqlite.untilReady;
+        // TODO: Replace with partial loads of space.
+
+        return new Promise(() => {
+          name: "test";
+        });
+      },
+      uploadToPds: async (bytes, opts) => {
+        return this.client.uploadToPDS(bytes, opts);
+      },
+      addClient: async (port) => this.connectMessagePort(port),
+      pauseSubscription: async (streamId) => {
+        // await this.openSpacesMaterializer?.pauseSubscription(streamId);
+      },
+      unpauseSubscription: async (streamId) => {
+        // await this.openSpacesMaterializer?.unpauseSubscription(streamId);
+      },
+      resolveHandleForSpace: async (spaceId, handleAccountDid) =>
+        this.client.resolveHandleForSpace(spaceId, handleAccountDid),
+      resolveSpaceFromHandleOrDid: async (handleOrDid) =>
+        this.client.resolveSpaceFromHandleOrDid(handleOrDid),
+      createStreamHandleRecord: async (spaceId) => {
+        await this.client.createStreamHandleRecord(spaceId);
+      },
+      removeStreamHandleRecord: async () => {
+        await this.client.removeStreamHandleRecord();
+      },
+    };
+  }
+
+  private logout() {
+    if (this.#auth.state === "authenticated") {
+      this.#auth.client.logout();
+      this.#auth = { state: "unauthenticated" };
+    } else {
+      console.warn("Already logged out");
+    }
+  }
+}
+
+type SqlitePending = {
+  state: "pending";
+};
+
+type SqliteReady = {
+  state: "ready";
+  proxy: SqliteWorkerInterface;
+};
+
+type SqliteState = SqlitePending | SqliteReady;
+
+class SqliteSupervisor {
+  #state: SqliteState;
+  #ready = new Deferred();
+  liveQueries: Map<string, { port: MessagePort; statement: SqlStatement }>;
+
+  constructor() {
+    this.#state = { state: "pending" };
+    this.liveQueries = new Map();
+  }
+
+  get untilReady() {
+    return this.#ready.promise;
+  }
+
+  get ready() {
+    return this.#state.state === "ready";
+  }
+
+  get proxy() {
```

Comment:
The name `proxy` here is kind of ambiguous to me. Something more like `sqlite` or `sqliteWorker` would probably be clearer.